Library/Homebrew/dev-cmd/boneyard-formula-pr.rb

```diff
@@ -0,0 +1,135 @@
+# Creates a pull request to boneyard a formula.
+#
+# Usage: brew boneyard-formula-pr [options...] <formula-name>
+#
+# Options:
+#   --dry-run:  Print what would be done rather than doing it.
+
+require "formula"
+require "utils/json"
+require "fileutils"
+
+begin
+  require "json"
+rescue LoadError
+  puts "Homebrew does not provide Ruby dependencies; install with:"
+  puts "  gem install json"
+  odie "Dependency json is not installed."
+end
+
+module Homebrew
+  def boneyard_formula_pr
+    formula = ARGV.formulae.first
+    odie "No formula found!" unless formula
+
+    formula_path = formula.path
+    formula_file = "#{formula.name}.rb"
+    bottle_block = File.read(formula_path).include? "  bottle do"
+    boneyard_tap = Tap.fetch("homebrew", "boneyard")
+    tap_migrations_path = formula.tap.path/"tap_migrations.json"
+    if ARGV.dry_run?
+      puts "brew update"
+      puts "brew tap #{boneyard_tap.name}"
+      puts "cd #{formula.tap.path}"
+      cd formula.tap.path
+      puts "cp #{formula_path.relative_path_from(formula.tap.path)} #{boneyard_tap.formula_dir}"
+      puts "git rm #{formula_path.relative_path_from(formula.tap.path)}"
+      unless File.exist? tap_migrations_path
+        puts "Creating tap_migrations.json for #{formula.tap.name}"
+        puts "git add #{tap_migrations_path}"
+      end
+      puts "Loading tap_migrations.json"
+      puts "Adding #{formula.name} to tap_migrations.json"
+    else
+      safe_system HOMEBREW_BREW_FILE, "update"
+      safe_system HOMEBREW_BREW_FILE, "tap", boneyard_tap.name
+      cd formula.tap.path
+      cp formula_path.relative_path_from(formula.tap.path), boneyard_tap.formula_dir
+      safe_system "git", "rm", formula_path.relative_path_from(formula.tap.path)
+      unless File.exist? tap_migrations_path
+        tap_migrations_path.write <<-EOS.undent
+          {
+          }
+        EOS
+        safe_system "git", "add", tap_migrations_path
+      end
+      tap_migrations = Utils::JSON.load(File.read(tap_migrations_path))
+      tap_migrations[formula.name] = boneyard_tap.name
+      tap_migrations = tap_migrations.sort.inject({}) { |a, e| a.merge!(e[0] => e[1]) }
+      tap_migrations_path.atomic_write(JSON.pretty_generate(tap_migrations) + "\n")
+    end
+    unless Formula["hub"].any_version_installed?
+      if ARGV.dry_run?
+        puts "brew install hub"
+      else
+        safe_system HOMEBREW_BREW_FILE, "install", "hub"
+      end
+    end
+    branch = "#{formula.name}-boneyard"
+    puts "cd #{formula.tap.formula_dir}" if ARGV.dry_run?
+    cd formula.tap.formula_dir
+    if ARGV.dry_run?
+      puts "git checkout -b #{branch} origin/master"
+      puts "git commit --no-edit --verbose --message=\"#{formula.name}: migrate to boneyard\" -- #{formula_path} #{tap_migrations_path}"
```

Comment:
This would be more readable if it used relative paths instead of absolute paths (but in this case it might be undesirable to `cd` to `formula.tap.formula_dir` and `formula.tap.path` might be more appropriate).